src/index.ts

```diff
@@ -119,11 +119,12 @@ export class Marqeta {
     for (let cnt = 0; cnt < 3; cnt++) {
       // now we can make the call... see if it's a JSON body or a FormData one...
       try {
+        const fieldsKey = method + ':' + uri
         response = await fetch(url, {
           method: method,
           body: isForm ?
             (body as any) :
-            (body ? JSON.stringify(snakecaseKeys(removeEmpty(cleanMetaData(body)))) : undefined),
+            (body ? JSON.stringify(snakecaseKeys(removeEmpty(cleanMetaData(body, fieldsKey)))) : undefined),
```

Comment:
OK, so if you retry this call you'll re-remove the data? Why not just do it once?

For example, what if, after:

```typescript
    if (!isForm) {
      headers = { ...headers, 'Content-Type': 'application/json' }
      // clean out the fields in the JSON body that could cause trouble
      body = cleanMetaData(body, method, uri)
    }
```

Then, within the `fetch()` call it's just:

```typescript
      try {
         response = await fetch(url, {
           method: method,
           body: isForm ?
             (body as any) :
             (body ? JSON.stringify(snakecaseKeys(removeEmpty(cleanMetaData(body)))) : undefined),
```

like before. It's one extra line in `fire()`... just the one - and the comment.

But at this point, I think `cleanMetaData()` is a bad name... is should be `cleanRequestBody()` - because that's really what it's doing... on the way **_to_** Marqeta. Right?

So then, you have:

```typescript
function cleanRequestBody(body, method, uri) {
}
```

Please note that this is **_not_** `export`-ed because this is something that is only going to be used in this module, and it's existence is not something anyone else needs to know about.

I'm going to comment on this function later... but for the `fire()` function, all we need to do is:

```typescript
    if (!isForm) {
      headers = { ...headers, 'Content-Type': 'application/json' }
      // clean out the fields in the JSON body that could cause trouble
      body = cleanRequestBody(body, method, uri)
    }
```

and then:

```typescript
      try {
         response = await fetch(url, {
           method: method,
           body: isForm ?
             (body as any) :
             (body ? JSON.stringify(snakecaseKeys(removeEmpty(body))) : undefined),
```

but honestly, since we're doing all this on the `body` already, why not have `removeEmpty()` within the `cleanRequestBody()` as that's part of the cleaning process... Right?

So we could have:

```typescript
  /*
   * These are the problem fields for sending to each of the indicated
   * method:ure pairs in the keys. These are something that the caller
   * can provide, but we need to filter out because Marqeta will error
   * out if we pass them.
   */
  const problemFields = {
    'POST:usertransitions': ['createdTime', 'lastModifiedTime', 'password'],
  }

  /*
   * Marqeta is picky about the JSON body data that we send on some calls,
   * and in order to make it possible for the user to pass in more than
   * is needed, we need to file out the problematic fields - based on the
   * method and uri that's being called. This function does that.
   */
  function cleanRequestData(obj: any, method: string, uri: string): any {
    const key = method + ':' + uri
    const defaultFields = ['createdTime', 'lastModifiedTime', 'password', 'status']
    (problemFields[key] || ['createdTime', 'lastModifiedTime', 'password', 'status'])
      .forEach(f => delete obj[f])
    return obj
  }
```

Now we have separated the definition of the `problemFields` and the simpler function where we decide what the key is... you don't want to let the implementation _leak_ out of the function... pass in the `method` and the `uri`, and the body, and get a clean `body`.

Now for that `removeEmpty()`... We can add:

```typescript
  /*
   * Marqeta is picky about the JSON body data that we send on some calls,
   * and in order to make it possible for the user to pass in more than
   * is needed, we need to file out the problematic fields - based on the
   * method and uri that's being called. This function does that.
   */
  function cleanRequestData(obj: any, method: string, uri: string): any {
    const key = method + ':' + uri
    const defaultFields = ['createdTime', 'lastModifiedTime', 'password', 'status']
    (problemFields[key] || ['createdTime', 'lastModifiedTime', 'password', 'status'])
      .forEach(f => delete obj[f])
    return snakecaseKeys(removeEmpty(obj))
  }
```

and now we can let the `fetch()` become:

```typescript
      try {
         response = await fetch(url, {
           method: method,
           body: isForm ?
             (body as any) :
             (body ? JSON.stringify(body) : undefined),
```

Isn't that simpler?